salt/modules/yumpkg.py

```diff
@@ -658,13 +637,29 @@ def list_upgrades(refresh=True, **kwargs):
     exclude_arg = _get_excludes_option(**kwargs)
 
     if salt.utils.is_true(refresh):
-        refresh_db(**kwargs)
+        refresh_db(check_update=False, **kwargs)
+
+    cmd = [_yum(), '--quiet']
+    cmd.extend(repo_arg)
+    cmd.extend(exclude_arg)
+    cmd.extend(['list', 'updates'])
+    out = __salt__['cmd.run_all'](cmd,
+                                  output_loglevel='trace',
+                                  ignore_retcode=True,
+                                  python_shell=False)
+    if out['retcode'] != 0 and 'Error:' in out:
+        return {}
+    # Find end of first line so we can skip it
+    header_end = out['stdout'].find('\n')
+    if header_end == -1:
+        return {}
 
-    repoquery_args = repo_arg + exclude_arg
-    repoquery_args.extend(['--all', '--pkgnarrow=updates'])
-    updates = _repoquery_pkginfo(repoquery_args)
+    updates = _yum_pkginfo(out['stdout'][header_end + 1:])
     return dict([(x.name, x.version) for x in updates])
 
+# Preserve expected CLI usage (yum list updates)
+list_updates = list_upgrades
```

Comment:
I don't think so, unless I'm misunderstanding how the ``__func_alias__`` works. As I understand, it is used so that a function that would normally shadow a builtin can be named something like ``zip_`` and then be interpreted by the loader as the desired function name (i.e. ``modulename.zip``. In this case, we already have a ``list_upgrades`` function, but I wanted to make sure that ``list_updates`` also works as YUM users are used to running ``yum list updates`` and may just try running ``pkg.list_updates`` intuitively.